### PR TITLE
Mp b2b product offer availabilities

### PR DESCRIPTION
**Pull request by kostiantynlitvinov**
## PR Description

TBD

## Checklist
- [x] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: tests/api/mp_b2b/glue/product_offer_endpoints/product_offers/negative.robot [hunk 1]
```diff
@@ -0,0 +1,49 @@
+*** Settings ***
+Suite Setup       SuiteSetup
+Test Setup    TestSetup
+Resource    ../../../../../../resources/common/common_api.robot
+Default Tags    glue
+
+*** Test Cases ***
```
[the remark this answers is not in the document]
Comment (kostiantynlitvinov) [reply tests/api/mp_b2b/glue/product_offer_endpoints/product_offers/negative.robot]: removed

File: resources/environments/environments_api_mp_b2b.json [hunk 2]
```diff
@@ -86,27 +86,27 @@
...
-      "currency_symbol_chf": "CHF",
-
-      "gross_mode": "GROSS_MODE",
-      "net_mode": "NET_MODE",
-
-      "locale_name_DE": "de_DE",
-      "locale_code_DE": "de",
-      "locale_name_EN": "en_US",
-      "locale_code_EN": "en"
+        "currency_name_eur": "Euro",
```
[the remark this answers is not in the document]
Comment (kostiantynlitvinov) [reply resources/environments/environments_api_mp_b2b.json]: changes reverted

**Review by OKrivtsova — DISMISSED** [review]

File: tests/api/mp_b2b/glue/product_offer_endpoints/product_offer_availabilities/negative.robot [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ENABLER
+    TestSetup
+
+Get_not_existing_product_offer_availabilities
+    When I send a GET request:    /product-offers/test/product-offer-availabilities
+    Then Response status code should be:    404
+    And Response reason should be:    Not Found
+    And Response should return error message:    Not Found
+
+Get_empty_product_offer_availabilities
```
Comment (OKrivtsova) [inline tests/api/mp_b2b/glue/product_offer_endpoints/product_offer_availabilities/negative.robot:17]: there are more fails than bugs in comments
image

File: tests/api/mp_b2b/glue/product_offer_endpoints/product_offer_availabilities/positive.robot [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+Test Setup    TestSetup
+Resource    ../../../../../../resources/common/common_api.robot
+Default Tags    glue
+
+*** Test Cases ***
+ENABLER
+    TestSetup
+
+# There is a bug MP-6749
+Get_product_offer_availabilities
```
Comment (OKrivtsova) [inline tests/api/mp_b2b/glue/product_offer_endpoints/product_offer_availabilities/positive.robot:12]: why do we have includes here right away and why don't we have checks for the includes (I know where is a bug but we can add checks regardless)
please move includes into a separate tests, add checks (and add bug there in the comments as well

File: tests/api/mp_b2b/glue/product_offer_endpoints/product_offer_availabilities/positive.robot [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    When I send a GET request:    /product-offers/${second_offer_with_vp}/product-offer-availabilities?include=product-offer-prices,product-offers,merchants
+    Then Response status code should be:    200
+    And Response reason should be:    OK
+    And Response header parameter should be:    Content-Type    ${default_header_content_type}
+    And Response body parameter should be:    [data][0][id]    ${second_offer_with_vp}
+    And Response body parameter should be:    [data][0][type]    product-offer-availabilities
```
Comment (OKrivtsova) [inline tests/api/mp_b2b/glue/product_offer_endpoints/product_offer_availabilities/positive.robot:18]: we are checking an array - we should use keywords for arrays where we can

Comment (kostiantynlitvinov) [reply tests/api/mp_b2b/glue/product_offer_endpoints/product_offer_availabilities/positive.robot:18]: replaced by array check

File: tests/api/mp_b2b/glue/product_offer_endpoints/product_offer_availabilities/positive.robot (hunk 4 above)
Comment (kostiantynlitvinov) [reply tests/api/mp_b2b/glue/product_offer_endpoints/product_offer_availabilities/positive.robot:12]: I checked once more. There are no includes in the endpoint response so I just remove the test with includes

File: tests/api/mp_b2b/glue/product_offer_endpoints/product_offer_availabilities/negative.robot (hunk 3 above)
Comment (kostiantynlitvinov) [reply tests/api/mp_b2b/glue/product_offer_endpoints/product_offer_availabilities/negative.robot:19]: review the tests, re-run them on the working environment, partially re-write them
now all of them should fail on the https://www.de.mp-b2b.internal-testing.demo-spryker.com/
Screenshot 2022-04-26 at 23 43 34
screen from the local env where os no bug

File: tests/api/mp_b2b/glue/product_offer_endpoints/product_offer_availabilities/positive.robot (hunk 5 above)
Comment (OKrivtsova) [reply tests/api/mp_b2b/glue/product_offer_endpoints/product_offer_availabilities/positive.robot:18]: I still see hardcoded [0] here. only one check below checks array
we have keywords like this
Each array element of array in response should contain property with value in:

Comment (kostiantynlitvinov) [reply tests/api/mp_b2b/glue/product_offer_endpoints/product_offer_availabilities/positive.robot]: did my best to avoid hardcoded indexes

**Review by OKrivtsova — APPROVED** [review]

File: resources/environments/environments_api_mp_b2b.json [hunk 6]
```diff
@@ -367,8 +366,6 @@
         "second_offer_with_volume_price": "offer9",
         "offer_with_volume_price": "offer10",
         "offer_with_volume_price_gross_net_prices": "offer48",
-        "merchant_office_king_id": "MER000009",
-        "merchant_budget_stationery_id": "MER000010",
```
Comment (OKrivtsova) [inline resources/environments/environments_api_mp_b2b.json:371]: why are these removed? they are be needed in other tests

Comment (kostiantynlitvinov) [reply resources/environments/environments_api_mp_b2b.json:371]: they are duplicated in the merchant section so were removed from products offer

**Review by OKrivtsova — APPROVED** [review]
